.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,117 @@
+# How to Contribute
```

Comment:
Thanks for the original pass at this @davsfa! I simplified the guidelines here a bit (and expanded them in other places). By its nature `msgspec` will never be a good project for Python beginners to contribute to - as such I dropped explanations that would only be useful for those new at Python and OSS development.

Following the ["write for one person"](https://wizardzines.com/comics/write-for-one-person/) idea, this document is targeted at someone who has contributed on GitHub before, and has some experience with Python development workflows.